pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -144,10 +155,34 @@ func (r *SubnetPortReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		WithOptions(
 			controller.Options{
 				MaxConcurrentReconciles: runtime.NumCPU(),
-				}).
+			}).
+		Watches(&source.Kind{Type: &vmv1alpha1.VirtualMachine{}},
+				handler.EnqueueRequestsFromMapFunc(r.vmMapFunc),
+				builder.WithPredicates(predicate.LabelChangedPredicate{})).
 		Complete(r) // TODO: watch the virtualmachine event and update the labels on NSX subnet port.
 }
 
+func (r *SubnetPortReconciler) vmMapFunc(vm client.Object) []reconcile.Request {
+	subnetPortList := &v1alpha1.SubnetPortList{}
+	// There's an assumption that the subnetport CR and its VirtualMachine are in the same namespace.
+	r.Client.List(context.TODO(), subnetPortList, client.InNamespace(vm.GetNamespace()))
+	var requests []reconcile.Request
+	for _, subnetPort := range subnetPortList.Items {
+		if subnetPort.Spec.AttachmentRef.Name == "" {
```

Comment:
since line 174 compare subnetPort.Spec.AttachmentRef.Name with vm.GetName(), it seems that line 171 not needed

Reply:
done, thanks